src/components/Icons/Arrow/interfaces.ts

```diff
@@ -1,25 +1,4 @@
-import { HTMLAttributes } from 'react';
+import { SVGProps } from 'react';
+import { IIconWithDirection, IIconWithSizes } from '../interfaces';
 
-interface SVGProps {
-  fill?: string;
-  width?: string;
-  height?: string;
-}
-
-export interface IArrowColors {
-  activeColor?: string;
-  defaultColor?: string;
-  inactiveColor?: string;
-}
-
-export interface IArrowCommonProps {
-  isActive?: boolean;
-  isInactive?: boolean;
-  direction?: 'up' | 'down' | 'right' | 'left';
-}
-
-export interface IArrow extends IArrowCommonProps {
-  colors?: IArrowColors;
-}
-
-export interface IArrowProps extends SVGProps, IArrowColors, IArrowCommonProps, HTMLAttributes<SVGSVGElement> {}
+export type IArrowProps = Omit<SVGProps<SVGSVGElement>, 'ref'> & IIconWithDirection & IIconWithSizes;
```

Comment:
I think it is important to keep the SVG props because we spread them in the svg, it allows us to have more flexibility when using the icons.

Reply:
since the use case for setting any svg prop besides color (and possibly className) is really not that common, how about placing the other svg props in a different field in order to make the icon interface more clear? e.g:
```ts
interface IIcon {
  color?: string;
  className?: string;
  svgProps: React.SVGProps<SVGSVGElement>;
}
```
then even doing auto-completion will be easier with icons having different props such as the Arrow

Reply:
Generally, I would advise to minimise the API that you're exposing, it will usually lead to variation in the UI that we should not have. 
That said I'm fine leaving this completely exposed and see how it progresses.